pkg/utils/common.go

```diff
@@ -184,16 +185,63 @@ func GetGPUConfigBySKU(instanceType string) (*sku.GPUConfig, error) {
 	return skuHandler.GetGPUConfigBySKU(instanceType), nil
 }
 
-func TryGetGPUConfigFromNode(ctx context.Context, kubeClient client.Client, workerNodes []string) (*sku.GPUConfig, error) {
-	skuGPUCount, err := FetchGPUCountFromNodes(ctx, kubeClient, workerNodes)
-	if err != nil || skuGPUCount == 0 {
-		return nil, fmt.Errorf("failed to fetch GPU count from nodes: %w", err)
+func TryGetGPUConfigFromNodes(ctx context.Context, workerNodes []*corev1.Node) (*sku.GPUConfig, error) {
+	if len(workerNodes) == 0 {
+		return nil, fmt.Errorf("no worker nodes found")
 	}
 
+	// Get the first node to extract GPU configuration
+	for _, node := range workerNodes {
+		// Try to get GPU configuration from nvidia.com labels first
+		if gpuConfig, err := GetGPUConfigFromNvidiaLabels(node); err == nil {
+			return gpuConfig, nil
+		}
+	}
+
+	// Fallback to the previous implementation using only GPU count
+	skuGPUCount := GetPerNodeGPUCountFromNodes(workerNodes)
+	if skuGPUCount == 0 {
+		return nil, fmt.Errorf("failed to fetch GPU count from nodes")
+	}
+
+	return &sku.GPUConfig{
+		SKU:       "unknown", // SKU is not available from nodes
+		GPUCount:  skuGPUCount,
+		GPUModel:  "unknown", // GPU model is not available from nodes
+		GPUMemGiB: 0,         // GPU memory is not available from capacity
+	}, nil
+}
+
+// GetGPUConfigFromNvidiaLabels extracts GPU configuration from nvidia.com labels on a node
+func GetGPUConfigFromNvidiaLabels(node *corev1.Node) (*sku.GPUConfig, error) {
+	gpuProduct, hasGPUProduct := node.Labels[consts.NvidiaGPUProduct]
+	gpuCountStr, hasGPUCount := node.Labels[consts.NvidiaGPUCount]
+	gpuMemoryStr, hasGPUMemory := node.Labels[consts.NvidiaGPUMemory]
+
+	// Check if all required nvidia.com labels are present
+	if !hasGPUProduct || !hasGPUCount || !hasGPUMemory {
+		return nil, fmt.Errorf("missing required nvidia.com labels on node %s", node.Name)
+	}
+
+	// Parse GPU count
+	gpuCount, err := strconv.Atoi(gpuCountStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid nvidia.com/gpu.count value on node %s: %s", node.Name, gpuCountStr)
+	}
+
+	// Parse GPU memory (nvidia.com/gpu.memory is in MiB, convert to GB)
+	gpuMemoryMiB, err := strconv.Atoi(gpuMemoryStr)
+	if err != nil {
```

Comment:
The err msg is ignored in the caller side, maybe we can print a log for this and the following conversion error cases.